grpcbp/prometheus.go

```diff
@@ -100,3 +102,16 @@ var (
 		Help: "The number of in-flight requests",
 	}, clientActiveRequestsLabels)
 )
+
+// serviceAndMethodSlug splits the UnaryServerInfo.FullMethod and returns
+// the package.service part separate from the method part.
+// ref: https://pkg.go.dev/google.golang.org/grpc#UnaryServerInfo
+func serviceAndMethodSlug(fullMethod string) (service string, method string) {
+	split := strings.SplitN(fullMethod, "/", 3)
+	if len(split) < 2 {
+		return "", ""
+	}
+	method = split[len(split)-1]
+	service = strings.Join(split[:len(split)-1], "")
+	return service, method
+}
```

Comment:
```
func serviceAndMethodSlug(fullMethod string) (service, method string) {
	split := strings.SplitN(fullMethod, "/", 3)
	if len(split) < 3 || split[0] != "" {
		return "", fullMethod
	}
	return split[1], split[2]
}
```

or

```
func serviceAndMethodSlug(fullMethod string) (service, method string) {
	fullMethod = strings.TrimPrefix(fullMethod, "/")
	split := strings.SplitN(fullMethod, "/", 2)
	if len(split) < 2 {
		return "", fullMethod
	}
	return split[0], split[1]
}
```

I kinda like the second one.

Reply:
Put differently, I think the static indices are easier to reason about than len-1 / len-2 -- and checking `len(split) < 2` when you have `SplitN(..., 3)` looks like an off-by-one error, even though it's not.  (it took me awhile to recognize this)

Reply:
agree, i like the second one. will add.